src/stdatamodels/jwst/datamodels/tests/test_multislit.py

```diff
@@ -105,27 +107,27 @@ def test_multislit_metadata():
 def test_multislit_metadata2():
     with MultiSlitModel() as ms:
         ms.slits.append(ms.slits.item())
-        for key, val in ms.items():
+        for _, val in ms.items():
             assert isinstance(val, (bytes, str, int, float, bool, Time))
 
 
 def test_multislit_copy(tmp_path):
     path = tmp_path / "multislit.fits"
-    with MultiSlitModel() as input:
-        for i in range(4):
-            input.slits.append(input.slits.item(data=np.empty((50, 50))))
+    with MultiSlitModel() as input_file:
+        for _ in range(4):
+            input_file.slits.append(input_file.slits.item(data=np.empty((50, 50))))
 
-        assert len(input.slits) == 4
-        input.save(path)
+        assert len(input_file.slits) == 4
+        input_file.save(path)
 
-        output = input.copy()
+        output = input_file.copy()
         assert len(output.slits) == 4
 
     with fits.open(path, memmap=False) as hdulist:
         assert len(hdulist) == 6
 
     with MultiSlitModel(path) as model:
-        for i, slit in enumerate(model.slits):
+        for i, _ in enumerate(model.slits):  # noqa: B007
             pass
         assert i + 1 == 4
```

Comment:
it isn't immediately clear to me what these lines are supposed to be doing, but I assume there is a reason for this?

Reply:
I see this as a bug in the rule, maybe we should disable it.

`i` is in scope and used on 132. Having a rule that suggests removing it would break the code.

Reply:
So far this rule has caught several of what I would consider "real" bad style in the various JWST PRs, so I'm in favor of keeping it.

You have to admit this is an unusual case, where there's nothing at all inside the `for` loop.  I'd suggest leaving it as-is

Reply:
Thanks! Yeah I say we keep the rule enabled.